Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialTypeSourceData.cpp

```diff
@@ -112,89 +127,351 @@ namespace AZ
+                if (propertyGroup->m_name == parsedPropertyId[0])
                 {
-                    if (action.m_operation == "rename")
+                    AZStd::span<const AZStd::string_view> subPath {parsedPropertyId.begin() + 1, parsedPropertyId.end()};
+
+                    if (subPath.size() == 1)
                     {
-                        if (action.m_renameFrom == propertyId.GetStringView())
+                        for (AZStd::unique_ptr<PropertyDefinition>& property : propertyGroup->m_properties)
                         {
-                            propertyId = MaterialPropertyId::Parse(action.m_renameTo);
-                            renamed = true;
+                            if (property->GetName() == subPath[0])
+                            {
+                                return property.get();
+                            }
                         }
                     }
-                    else
+                    else if(subPath.size() > 1)
                     {
-                        AZ_Warning("Material source data", false, "Unsupported material version update operation '%s'", action.m_operation.c_str());
+                        const MaterialTypeSourceData::PropertyDefinition* property = FindProperty(subPath, propertyGroup->m_propertyGroups);
+                        if (property)
+                        {
+                            return property;
+                        }
                     }
                 }
             }
 
-            return renamed;
+            return nullptr;
+        }
+        
+        MaterialTypeSourceData::PropertyDefinition* MaterialTypeSourceData::FindProperty(AZStd::span<AZStd::string_view> parsedPropertyId, AZStd::span<AZStd::unique_ptr<PropertyGroup>> inPropertyGroupList)
+        {
+            return const_cast<MaterialTypeSourceData::PropertyDefinition*>(const_cast<const MaterialTypeSourceData*>(this)->FindProperty(parsedPropertyId, inPropertyGroupList));
+        }
+
+        const MaterialTypeSourceData::PropertyDefinition* MaterialTypeSourceData::FindProperty(AZStd::string_view propertyId) const
+        {
+            AZStd::vector<AZStd::string_view> tokens = TokenizeId(propertyId);
+            return FindProperty(tokens, m_propertyLayout.m_propertyGroups);
+        }
+        
+        MaterialTypeSourceData::PropertyDefinition* MaterialTypeSourceData::FindProperty(AZStd::string_view propertyId)
+        {
+            AZStd::vector<AZStd::string_view> tokens = TokenizeId(propertyId);
+            return FindProperty(tokens, m_propertyLayout.m_propertyGroups);
+        }
+
+        AZStd::vector<AZStd::string_view> MaterialTypeSourceData::TokenizeId(AZStd::string_view id)
+        {
+            AZStd::vector<AZStd::string_view> tokens;
+
+            AzFramework::StringFunc::TokenizeVisitor(id, [&tokens](AZStd::string_view t)
+                {
+                    tokens.push_back(t);
+                },
+                "./", true, true);
```

Comment:
why is `"./"` instead of `"."`?

The comment in the header indicates `"itemA.itemB.itemC" into a vector like ["itemA", "itemB", "itemC"].` Being dot the separator.